code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -8,20 +14,39 @@
 	var/projectiletype
 	var/projectilesound
 	var/casingtype
-
+	var/ranged_cooldown = 0 //What the current cooldown on ranged attacks is, generally world.time + ranged_cooldown_time
 	var/list/friends = list()
 	var/break_stuff_probability = 10
+	var/ranged_ignores_vision
 	stop_automated_movement_when_pulled = 0
 	var/destroy_surroundings = 1
 	a_intent = I_HURT
 	can_burrow = TRUE
 	hunger_enabled = 0//Until automated eating mechanics are enabled, disable hunger for hostile mobs
+	var/minimum_distance = 1 //Minimum approach distance, so ranged mobs chase targets down, but still keep their distance set in tiles
+	var/atom/targets_from = null //all range/attack/etc. calculations should be done from this atom, defaults to the mob itself, useful for Vehicles and such
+	var/vision_range = 9 //How big of an area to search for targets in, a vision of 9 attempts to find targets as soon as they walk into screen view
+	var/aggro_vision_range = 9 //If a mob is aggro, we search in this radius. Defaults to 9 to keep in line with original simple mob aggro radius
+	var/approaching_target = FALSE //We should dodge now
+
+/mob/living/simple_animal/hostile/proc/get_view_range()
+	var/valueofview = vision_range
+	if(istype(src, /mob/living/simple_animal/hostile/megafauna/one_star))
```

Comment:
this is a fucking terrible thing to do and I hadn't seen it being done outside of ancient code.
I saw it's done in MoveToTarget but it's trickier to properly split proc there so fine, as long as it works.
But here it's completely unncessary, this shit must be done by overriding proc in subtypes, NOT typechecking youself ot be subtype just in case.

So what you do is have

```
/mob/living/simple_animal/hostile/proc/get_view_range()
	return vision_range
```
and in one star boss have
```
/mob/living/simple_animal/hostile/megafauna/one_star/get_view_range()
		if(istype(loc, /turf))
			var/turf/TURF = loc
			if(TURF.get_lumcount() > 0)
				return  10
			else
				return  4
		else
			return 0